android/src/com/appc/titanium/socketio/TiSocketioModule.java

```diff
@@ -51,11 +52,72 @@ public static void onAppCreate(TiApplication app)
 	public SocketIOClientProxy connect(String uri, @Kroll.argument(optional=true) HashMap options) throws URISyntaxException
 	{
 		Options socketOptions = new Options();
+		boolean autoConnect = true;
+		if (options != null) {
+			HashMap<String, Object> jsOptions = new HashMap<String, Object>(options);
+
+			autoConnect = (boolean)jsOptions.getOrDefault("autoConnect", true);
+			jsOptions.remove("autoConnect");
+			jsOptions.remove("parser");
 
+			this.applyOptions(jsOptions, socketOptions);
+		}
+
+		Log.d(LCAT, String.format("Connecting to socket on %s with options %s", uri, options));
 		Socket socket = IO.socket(uri, socketOptions);
-		socket.connect();
+		if (autoConnect) {
+			socket.connect();
+		}
+
+		return new SocketIOClientProxy(socket);
+	}
+
+	private void applyOptions(HashMap<String, Object> jsOptions, Options socketOptions)
```

Comment:
I would recommend this method to be change to return an instance of Options as a result. Something like that:
```
	private Options parseJSOptions(HashMap<String, Object> jsOptions) {
		Options socketOptions = new Options();
		for (Map.Entry<String, Object> entry: jsOptions.entrySet()) {
			String optionName = entry.getKey();
			Object optionValue = entry.getValue();

			if (optionName.equals("query") && optionValue instanceof HashMap) {
				optionValue = this.buildQueryString((HashMap<?, ?>)optionValue);
			}

			try {
				Field propertyField = socketOptions.getClass().getField(optionName);
				propertyField.set(socketOptions, optionValue);
			} catch (Exception e) {
				if (e instanceof NoSuchFieldException) {
					Log.w(TAG, String.format("There is no option named \"%s\".", optionName));
				} else if (e instanceof IllegalArgumentException) {
					Log.w(TAG, String.format("Invalid value for option \"%s\".", optionName));
				} else {
					Log.w(TAG, String.format("Error while trying to set socket option %s. Error: %s", optionName, e.getMessage()));
				}
			}
		}
		return socketOptions;
	}
```

And that result be passed to `IO.socket(uri, socketOptions)` above.